Quadriga/src/main/java/edu/asu/spring/quadriga/web/dictionary/DictionaryItemSearchAddController.java

```diff
@@ -96,29 +90,64 @@ public String addDictionaryItem(HttpServletRequest req, @PathVariable("dictionar
             IDictionary dictionary = dictionaryManager.getDictionaryDetails(dictionaryId);
             model.addAttribute("dictionary", dictionary);
             model.addAttribute("dictID", dictionaryId);
-            
+
             model.addAttribute("show_error_alert", true);
-            model.addAttribute("error_alert_msg", messageSource.getMessage("dictionary.items.add.no_selection", new Object[] {}, locale));
+            model.addAttribute("error_alert_msg",
+                    messageSource.getMessage("dictionary.items.add.no_selection", new Object[] {}, locale));
 
             return "auth/dictionaries/addDictionaryItems";
         }
-        
+
         dictionaryManager.addDictionaryItems(dictionaryItems, values, dictionaryId);
         model.addAttribute("show_success_alert", true);
-        model.addAttribute("success_alert_msg", messageSource.getMessage("dictionary.items.add.success", new Object[] {}, locale));
-        
+        model.addAttribute("success_alert_msg",
+                messageSource.getMessage("dictionary.items.add.success", new Object[] {}, locale));
+
         List<IDictionaryItems> dictionaryItemList = dictionaryManager.getDictionaryItems(dictionaryId);
-        String dictionaryName = dictionaryManager.getDictionaryName(dictionaryId);
         model.addAttribute("dictionaryItemList", dictionaryItemList);
-        model.addAttribute("dictName", dictionaryName);
-        model.addAttribute("dictID", dictionaryId);
-        IDictionary dictionary = dictionaryManager.getDictionaryDetails(dictionaryId);
-        model.addAttribute("dictionary", dictionary);
-        JSONObject core = new JSONObject();
-        model.addAttribute("core", core.toString());
+
+        IDictionary dictionaryObj = dictionaryManager.getDictionaryDetails(dictionaryId);
+
+        String userName = principal.getName();
+        boolean owner = dictionaryObj.getOwner().getUserName().equals(userName);
+
+        model.addAttribute("owner", owner);
+        if (owner) {
+            model.addAttribute("isAdmin", true);
+            model.addAttribute("hasWrite", true);
+            model.addAttribute("hasRead", true);
+        } else {
+            List<IDictionaryCollaborator> existingCollaborators = dictionaryManager
+                    .showCollaboratingUsers(dictionaryId);
+            setPermissions(model, userName, existingCollaborators);
+        }
+
```

Comment:
why don't you just redirect to the dictionary page with a success message instead of duplicating the code?